docs/DeveloperGuide.md

```diff
@@ -154,6 +154,47 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Sort Persons
+
+The sort persons feature allows the user to sort persons by their name, date of birth or gender. Sorting is performed as part of `listPersonsCommand#execute()`.
+
+The sorting operation is exposed in the `Model` interface as `Model#sortPersons()` which calls `AddressBook#sortPersons()` which in turn calls `UniquePersonList#sort()` to sort the underlying `ObservableList<Person>`.
+
+The following sequence diagram illustrates the relevant sorting method calls for the command `listPersons s/n` with a `SortField` object `s`:
+
+![SortPersonsSequenceDiagram](images/SortPersonsSequenceDiagram.png)
+
+
+The field to sort persons by is encapsulated within the `SortField` class. `SortField` stores the sort field enumeration (`SortFieldType`) as well as the `Comparator` object that defines the sorting logic.
+
+The following class diagram shows how the `SortField` class integrates with the other components:
+
+![SortFieldClassDiagram](images/SortFieldClassDiagram.png)
+
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Sorting is performed directly on the underlying `UniquePersonList` object, which means the sorted result is **permanent**. i.e. Not a temporary result for the `listPersons` command. For example, if `listPersons s/n` and `listPersons` are executed back-to-back, the result of the second `listPersons` command will still display the sorted results from the first `listPersons s/n` command because the sorted result is permanent.<br><br>
```

Comment:
```suggestion
<div markdown="span" class="alert alert-info">:information_source: **Note:** Sorting is performed directly on the underlying `UniquePersonList` object, which means the sorted result is **permanent**. For example, if `listPersons s/n` and `listPersons` are executed back-to-back, the result of the second `listPersons` command will display the sorted results from the first `listPersons s/n` command because the sorted result is permanent.<br><br>
```
I was thinking these changes might make the documentation less repetitive!